cmd/parser/parse.go

```diff
@@ -0,0 +1,356 @@
+	cobraCmd.Flags().BoolVarP(&cmd.noColor, "no-color", "n", false, "Disable use color characters when generating output.")
+	cobraCmd.MarkFlagsOneRequired("entry", "file")
+
+	return cobraCmd
+}
+
+func (cmd *ParseCommand) Run() error {
+	cmd.parser = &rules.SimpleParser{}
+
+	if cmd.noColor {
+		colorError = ""
+		colorOk = ""
+		colorNeutral = ""
+		colorMatched = ""
+	}
+
+	if cmd.match != "" && (!cmd.sort || !cmd.unique) {
+		cmd.cobraCmd.Printf("\tMatching is only supported when both priority and uniqnuess flags are specified.\n")
+		return nil
+	}
+	
+	if (cmd.useGrammar) {
+		cmd.parser = &grammar.GrammarParser{}
+	}
+
+	var entries []string
+	if cmd.ruleFilePath != "" {
+		conf := &conf{}
+		conf.getConf(cmd.ruleFilePath)
+		cmd.cobraCmd.Printf("Parsing rules from file: %s\n", cmd.ruleFilePath)
+		entries = conf.Rules
+	} else {
+		entries = strings.Split(cmd.rule, ";")
+	}
+
+	allResults := make([]rules.ParsingResult, 0, len(entries))
+	okResults := make([]rules.ParsingResult, 0, len(entries))
+	errorResults := make([]rules.ParsingResult, 0, len(entries))
+
+	resolvedRules := make(map[string]rules.ParsingResult)
+
+	for _, entry := range entries {
+		rule, err := cmd.parser.Parse(entry)
+
+		result := rules.ParsingResult{OriginalRule: entry,  Rule: rule, Err: err}
+		
+		if err != nil {
+			errorResults = append(errorResults, result)	
+		} else {
+			okResults = append(okResults, result)	
+			if rule.IsResolved() {
+				resolvedRules[rule.StringNoLabels()] = result
+			}
+		}
+
+		allResults = append(allResults, result)
+	}
+
+	if cmd.sort {
+		allResults = rules.SortRuleEntries(allResults)
+		okResults = rules.SortRuleEntries(okResults)
+		errorResults = rules.SortRuleEntries(errorResults)
+	}
+
+	if cmd.unique {
+		uniqnuessSet := make(map[string]rules.ParsingResult)
+		signatureSet := make(map[string]rules.ParsingResult)
+		uniqueResults := make([]rules.ParsingResult, 0, len(allResults))
+
+
+		for _, result := range allResults {
+
+			// containsWildcards := false
+
+			if result.Err != nil {
+				uniqueResults = append(uniqueResults, result)
+				continue
+			}
+
+			negativeSignatureKey := result.Rule.Plan
+			signatureKey := result.Rule.Plan
+			key := result.Rule.Plan
+			
+			key += "PR:" 
+			signatureKey += "PR:"
+			negativeSignatureKey += "PR:"
+			if result.Rule.PlatformRegion == "" || result.Rule.PlatformRegion == "*" {
+				key += "*"
+
+				signatureKey += "*"
+				negativeSignatureKey += "attr"
+				if result.Rule.PlatformRegion == "*" {
+					// containsWildcards = true
+				}
+			} else {
+				key += result.Rule.PlatformRegion
+			
+				signatureKey += "attr"
+				negativeSignatureKey += "*"
+			}
+			
+			key += "HR:"
+			signatureKey += "HR:"
+			negativeSignatureKey += "HR:"
+			if result.Rule.HyperscalerRegion == "" || result.Rule.HyperscalerRegion == "*" {
```

Comment:
Code duplication. We could possibly extract part of it as function and call it twice.